pkg/gitprovider/gitee.go

```diff
@@ -0,0 +1,380 @@
+	HumanName     string           `json:"human_name,omitempty"`
+	Url           string           `json:"url,omitempty"`
+	Namespace     *gitee.Namespace `json:"namespace,omitempty"`
+	Path          string           `json:"path,omitempty"`
+	Name          string           `json:"name,omitempty"`
+	Owner         *gitee.UserBasic `json:"owner,omitempty"`
+	Description   string           `json:"description,omitempty"`
+	Private       bool             `json:"private,omitempty"`
+	Public        bool             `json:"public,omitempty"`
+	Internal      bool             `json:"internal,omitempty"`
+	HtmlUrl       string           `json:"html_url,omitempty"`
+	Homepage      string           `json:"homepage,omitempty"`
+	DefaultBranch string           `json:"default_branch,omitempty"`
+	Parent        *Project         `json:"parent,omitempty"`
+}
+
+type GiteeGitProvider struct {
+	*AbstractGitProvider
+
+	token string
+}
+
+func NewGiteeGitProvider(token string) *GiteeGitProvider {
+	gitProvider := &GiteeGitProvider{
+		token:               token,
+		AbstractGitProvider: &AbstractGitProvider{},
+	}
+	gitProvider.AbstractGitProvider.GitProvider = gitProvider
+
+	return gitProvider
+}
+
+func (g *GiteeGitProvider) CanHandle(repoUrl string) (bool, error) {
+	staticContext, err := g.ParseStaticGitContext(repoUrl)
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains("https://gitee.com", staticContext.Source), nil
+}
+
+func (g *GiteeGitProvider) getApiClient() *gitee.APIClient {
+	tokenSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: g.token})
+	conf := gitee.NewConfiguration()
+	conf.HTTPClient = oauth2.NewClient(context.Background(), tokenSource)
+	return gitee.NewAPIClient(conf)
+}
+
+func (g *GiteeGitProvider) GetUser() (*GitUser, error) {
+	client := g.getApiClient()
+	ctx := context.Background()
+	user, _, err := client.UsersApi.GetV5User(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch User : %w", err)
+	}
+
+	gitUser := &GitUser{
+		Id:       strconv.FormatInt(int64(user.Id), 10),
+		Username: user.Login,
+		Name:     user.Name,
+	}
+
+	// gitee email will be empty if not set to public
+	if user.Email != "" {
+		gitUser.Email = user.Email
+	}
+
+	return gitUser, nil
+}
+
+func (g *GiteeGitProvider) GetNamespaces(options ListOptions) ([]*GitNamespace, error) {
+	client := g.getApiClient()
+	ctx := context.Background()
+	userNamespaces, _, err := client.UsersApi.GetV5UserNamespaces(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch Namespaces : %w", err)
+	}
+
+	var namespaces []*GitNamespace
+	for _, namespace := range userNamespaces {
+		namespaces = append(namespaces, &GitNamespace{
+			Id:   namespace.Name,
+			Name: namespace.Name,
+		})
+	}
+
+	return namespaces, nil
+}
+
+func (g *GiteeGitProvider) GetRepositories(namespace string, options ListOptions) ([]*GitRepository, error) {
+	reposApiURL := fmt.Sprintf("https://gitee.com/api/v5/user/repos?access_token=%s", g.token)
```

Comment:
go-gitee doesnt have a function that returns a list of projects/repositories, so I had to do it by making an HTTP request like this.